src/main/java/Parser.java

```diff
@@ -0,0 +1,84 @@
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.io.IOException;
+
+public class Parser {
+
+    public Task parse(InputHandler.CommandType type, String[] splitInput) throws DukeException {
```

Comment:
Perhaps it may be easier to understand if the 2 parse functions were combined into 1 :)